diff --git a/src/wannier_lib.F90 b/src/wannier_lib.F90
index 37f11f2..7395f45 100644
--- a/src/wannier_lib.F90
+++ b/src/wannier_lib.F90
@@ -56,3 +56,4 @@ subroutine wannier_setup(seed__name, mp_grid_loc, num_kpts_loc, &
                          proj_site_loc, proj_l_loc, proj_m_loc, proj_radial_loc, proj_z_loc, &
-                         proj_x_loc, proj_zona_loc, exclude_bands_loc, proj_s_loc, proj_s_qaxis_loc)
+                         proj_x_loc, proj_zona_loc, exclude_bands_loc, proj_s_loc, proj_s_qaxis_loc, &
+                         wb_out_, dis_win_min_out_, dis_win_max_out_, dis_froz_min_out_, dis_froz_max_out_, frozen_states_out_)
 
@@ -102,2 +103,7 @@ subroutine wannier_setup(seed__name, mp_grid_loc, num_kpts_loc, &
 
+  real(kind=dp), optional, intent(out) :: wb_out_(:)
+  real(kind=dp), optional, intent(out) :: dis_win_min_out_,dis_win_max_out_
+  real(kind=dp), optional, intent(out) :: dis_froz_min_out_,dis_froz_max_out_
+  logical, optional, intent(out) :: frozen_states_out_
+
   real(kind=dp) time0, time1, time2
@@ -161,2 +167,9 @@ subroutine wannier_setup(seed__name, mp_grid_loc, num_kpts_loc, &
 
+  if (present(wb_out_)) wb_out_(1:size(wb)) = wb(1:size(wb))
+  if (present(dis_win_min_out_)) dis_win_min_out_=dis_win_min
+  if (present(dis_win_max_out_)) dis_win_max_out_=dis_win_max
+  if (present(dis_froz_min_out_)) dis_froz_min_out_=dis_froz_min
+  if (present(dis_froz_max_out_)) dis_froz_max_out_=dis_froz_max
+  if (present(frozen_states_out_)) frozen_states_out_=frozen_states
+
   ! Now we zero all of the local output data, then copy in the data
